File: app/src/main/java/com/platdmit/simplecloudmanager/app/fragments/ServerFragment.java
package com.platdmit.simplecloudmanager.app.fragments;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;
import com.platdmit.simplecloudmanager.R;
import com.platdmit.simplecloudmanager.app.adapters.ServerViewStateAdapter;
import com.platdmit.simplecloudmanager.app.vm.ServerViewModel;
import com.platdmit.simplecloudmanager.app.vm.factory.SingleElementViewModelFactory;
import com.platdmit.simplecloudmanager.data.api.implement.ApiServerRepoImp;
import com.platdmit.simplecloudmanager.domain.SCMApp;
import com.platdmit.simplecloudmanager.domain.converters.implement.ServerConvertImp;
import com.platdmit.simplecloudmanager.domain.helpers.ContentUpdateService;
import com.platdmit.simplecloudmanager.domain.models.Server;
import com.platdmit.simplecloudmanager.domain.repo.ServerBaseRepo;
import com.platdmit.simplecloudmanager.domain.repo.implement.ServerRepoImp;
import com.platdmit.simplecloudmanager.domain.repo.implement.UpdateScheduleRepImp;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.viewpager2.widget.ViewPager2;

public class ServerFragment extends Fragment {
    private static final String TAG = ServerFragment.class.getSimpleName();

    private ServerViewModel mServerViewModel;

    private ImageView mServerImageLogo;
    private TextView mServerImageName;
    private TextView mServerIp;
    private TextView mServerStatus;
    private TextView mServerId;
    private TextView mServerPrice;
    private TextView mServerPaymentDate;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        if(savedInstanceState != null){
            mServerViewModel = new ViewModelProvider(this).get(ServerViewModel.class);
        } else {
            mServerViewModel = new ViewModelProvider(this,
                    new SingleElementViewModelFactory<ServerBaseRepo>(
                            new ServerRepoImp(
                                    new ApiServerRepoImp(SCMApp.getActualApiKeyService()),
                                    SCMApp.getDb(),
                                    new ServerConvertImp(),
                                    new ContentUpdateService(new UpdateScheduleRepImp(SCMApp.getDb()))
                            ), ServerBaseRepo.class, getArguments().getLong("ELEMENT_ID")
                    )).get(ServerViewModel.class);
        }

        View v = inflater.inflate(R.layout.fragment_server, container, false);

        mServerImageLogo = v.findViewById(R.id.server_image_logo);
        mServerImageName = v.findViewById(R.id.server_image_name);
        mServerIp = v.findViewById(R.id.server_ip);
        mServerStatus = v.findViewById(R.id.server_status);

        mServerId = v.findViewById(R.id.server_id);
        mServerPrice = v.findViewById(R.id.server_price);
        mServerPaymentDate = v.findViewById(R.id.server_paymentDate);

        mServerViewModel.getServerLiveData().observe(getViewLifecycleOwner(), this::initData);

        ServerViewStateAdapter serverViewPagerAdapter = new ServerViewStateAdapter.Builder()
                .add(new ServerTabMainFragment(getResources().getString(R.string.server_tab_main)))
                .add(new ServerTabActionsFragment(getResources().getString(R.string.server_tab_action)))
                .add(new ServerTabBackupsFragment(getResources().getString(R.string.server_tab_backups)))
                .add(new ServerTabStatisticsFragment(getResources().getString(R.string.server_tab_statistics)))
                .build(getChildFragmentManager(), getLifecycle(), getArguments().getLong("ELEMENT_ID"));


        ViewPager2 viewPager = v.findViewById(R.id.view_pager);
        viewPager.setAdapter(serverViewPagerAdapter);

        TabLayout tabLayout = v.findViewById(R.id.tabs);
        new TabLayoutMediator(tabLayout, viewPager, (tab, position) -> tab.setText(serverViewPagerAdapter.getPageTitle(position))).attach();

        return v;
    }

    @SuppressLint("SetTextI18n")
    private void initData(Server server){
        mServerImageName.setText(server.getName());
        mServerIp.setText(getResources().getString(R.string.server_ip_pref) + server.getV4Ip());
        mServerStatus.setText(server.getStatus());
        mServerId.setText(getResources().getString(R.string.server_id_pref) + server.getId());
        mServerPrice.setText(server.getPaymentPrice());
        mServerPaymentDate.setText(server.getPaymentDate());
        ((AppCompatActivity)getActivity()).getSupportActionBar().setTitle(server.getName());
    }
}

File: app/src/main/java/com/platdmit/simplecloudmanager/fragments/ServerFragment.kt
package com.platdmit.simplecloudmanager.fragments

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.Observer
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator
import com.google.android.material.tabs.TabLayoutMediator.TabConfigurationStrategy
import com.platdmit.simplecloudmanager.R
import com.platdmit.simplecloudmanager.adapters.ServerViewStateAdapter
import com.platdmit.simplecloudmanager.vm.ServerViewModel
import com.platdmit.domain.models.Server
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.android.synthetic.main.fragment_server.*

@AndroidEntryPoint
class ServerFragment : Fragment(R.layout.fragment_server) {
    private val serverViewModel: ServerViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if(savedInstanceState == null){
            serverViewModel.setActiveId(requireArguments().getLong("ELEMENT_ID"))
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        serverViewModel.serverLiveData.observe(viewLifecycleOwner, Observer { initData(it) })

        val serverViewPagerAdapter = ServerViewStateAdapter
                .add(ServerTabMainFragment(resources.getString(R.string.server_tab_main)))
                .add(ServerTabActionsFragment(resources.getString(R.string.server_tab_action)))
                .add(ServerTabBackupsFragment(resources.getString(R.string.server_tab_backups)))
                .add(ServerTabStatisticsFragment(resources.getString(R.string.server_tab_statistics)))
                .build(childFragmentManager, lifecycle, requireArguments().getLong("ELEMENT_ID"))

        serverViewPagerAdapter?.let {
            view_pager.adapter = it
            TabLayoutMediator(tabs, view_pager, TabConfigurationStrategy {
                tab: TabLayout.Tab, position: Int -> tab.text = it.getPageTitle(position)
            }).attach()
        }
     }

    @SuppressLint("SetTextI18n")
    private fun initData(server: Server) {
        server_image_name.text = server.name
        server_ip.text = resources.getString(R.string.server_ip_pref) + server.v4Ip
        server_status.text = server.status
        server_id.text = resources.getString(R.string.server_id_pref) + server.id
        server_price.text = server.paymentPrice
        server_paymentDate.text = server.paymentDate
        activity?.actionBar?.title = server.name
    }
}